aws/resource_aws_codebuild_project.go

```diff
@@ -659,6 +679,22 @@ func expandProjectEnvironment(d *schema.ResourceData) *codebuild.ProjectEnvironm
 		projectEnv.ImagePullCredentialsType = aws.String(v.(string))
 	}
 
+	if v := envConfig["registry_credential"]; v != nil {
+		config := v.(*schema.Set).List()[0].(map[string]interface{})
```

Comment:
With the switch to `TypeList` and the length check added above for safety, this simplifies to:

```suggestion
		config := v.([]interface{}).[0].(map[string]interface{})
```